app/helpers/sufia/sufia_helper_behavior.rb

```diff
@@ -154,8 +154,8 @@ def search_form_action
     end
 
     def render_visibility_link document
-      link_to render_visibility_label(document), sufia.edit_generic_file_path(document.noid, {anchor: "permissions_display"}),
-        id: "permission_"+document.noid, class: "visibility-link"
+      link_to render_visibility_label(document), sufia.edit_generic_file_path(document.id, {anchor: "permissions_display"}),
```

Comment:
but your'e using `id` in this path. Should it just be `document` as above in @cam156 comment?